Saas/New-BcEnvironment.ps1

```diff
@@ -28,96 +30,106 @@
 #>
 function New-BcEnvironment {
     Param(
-        [Parameter(Mandatory=$true)]
+        [Parameter(Mandatory = $true)]
         [Hashtable] $bcAuthContext,
         [string] $applicationFamily = "BusinessCentral",
-        [Parameter(Mandatory=$true)]
+        [Parameter(Mandatory = $true)]
         [string] $environment,
-        [Parameter(Mandatory=$true)]
+        [Parameter(Mandatory = $true)]
         [string] $countryCode,
-        [ValidateSet('Sandbox','Production')]
+        [ValidateSet('Sandbox', 'Production')]
         [string] $environmentType = "Sandbox",
         [string] $ringName = "PROD",
         [string] $applicationVersion = "",
         [string] $applicationInsightsKey = "",
         [switch] $doNotWait
     )
 
-$telemetryScope = InitTelemetryScope -name $MyInvocation.InvocationName -parameterValues $PSBoundParameters -includeParameters @()
-try {
+    $telemetryScope = InitTelemetryScope -name $MyInvocation.InvocationName -parameterValues $PSBoundParameters -includeParameters @()
+    try {
 
-    $bcAuthContext = Renew-BcAuthContext -bcAuthContext $bcAuthContext
-    if (Get-BcEnvironments -bcAuthContext $bcAuthContext | Where-Object { $_.Status -eq 'Preparing'}) {
-        Write-Host -NoNewline "Waiting for other environments."
-        while (Get-BcEnvironments -bcAuthContext $bcAuthContext | Where-Object { $_.Status -eq 'Preparing'}) {
-            Start-Sleep -Seconds 2
-            Write-Host -NoNewline "."
+        $bcAuthContext = Renew-BcAuthContext -bcAuthContext $bcAuthContext
+        if (Get-BcEnvironmentsOperations -bcAuthContext $bcAuthContext | Where-Object { ($_.productFamily -eq $applicationFamily) -and ($_.environmentName -eq $environment) -and ($_.status -in "queued", "scheduled", "running") }) {
+            Write-Host -NoNewline "Waiting for other environments."
+            while (Get-BcEnvironmentsOperations -bcAuthContext $bcAuthContext | Where-Object { ($_.productFamily -eq $applicationFamily) -and ($_.environmentName -eq $environment) -and ($_.status -in "queued", "scheduled", "running") }) {
```

Comment:
are we checking for other environments - then I guess we shouldn't check for $_.environmentName -eq $environmentName?

Reply:
Here I think that we do not need wait for other environments. We can wait for actual environment. So I will change text only.